src/ui/app/MfeContainer.js

```diff
@@ -16,10 +16,11 @@ const MfeContainer = ({ id, history }) => {
   const systemReport = useSelector(getSystemReport);
 
   useEffect(() => {
-    if (!window.appBuilderRouter) {
-      window.appBuilderRouter = history;
-    }
     const entandoWindow = window.entando || {};
+    if (!entandoWindow.router) {
+      window.entando = { ...entandoWindow, router: history };
+      entandoWindow.router = history;
```

Comment:
having to set the `window.entando` and the `entandoWindow` is a bit weird.
Maybe if we reorganize the code, we can create the `entandoWindow` as needed and then set it to the `window.entando` at the end just once

Reply:
agree, done ✅